src/Demo.java

```diff
@@ -0,0 +1,12 @@
+public class Demo {
+    public static void main(String[] args) {
+
+        Cat.allCatsResultOutput();
+        Dog.allDogsResultOutput();
+        allAnimalsResultOutput();
+    }
+
+    public static void allAnimalsResultOutput() {
```

Comment:
Sorry, but this is not the right way of implementing the animal counter. Would be better to declare a `private static int animalCounter' variable, and add a method `public static int getAnimalCounter()' that returns `animalCounter` internally. And please do not forget, that once an animal is created (the constructor of the class `Animal` is called) your `animalCounter` should be incremented. 
The same approach should be applied to the classes `Cat` and `Dog`.
Please think of it carefully and if you have questions, please ask before starting your implementation.